# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/libreoffice/hotfix-mips.patch
# Copyright (C) 2025 - 2026 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

CbcModel.hpp:2397:78: error: expected ',' or '...' before numeric constant
 2397 |     void setMIPStart( const std::vector< std::pair< std::string, double > > &mips ) {
      |                                                                              ^~~~
CbcModel.hpp: In member function 'void CbcModel::setMIPStart(const std::vector<std::pair<std::__cxx11::basic_string<char>, double> >&)':
CbcModel.hpp:2398:26: error: no match for 'operator=' (operand types are 'std::vector<std::pair<std::__cxx11::basic_string<char>, double> >' and 'int')
 2398 |        this->mipStart_ = mips;
      |                          ^~~~
CbcModel.hpp:2398:26: note: there are 3 candidates

--- libreoffice-25.8.0.4/external/coinmp/mips.patch.vanilla	2025-08-26 11:56:07.710349373 +0200
+++ libreoffice-25.8.0.4/external/coinmp/mips.patch	2025-08-26 11:55:48.467624339 +0200
@@ -0,0 +1,13 @@
+--- Cbc/src/CbcModel.hpp.vanilla	2025-08-26 11:31:00.587632497 +0200
++++ Cbc/src/CbcModel.hpp	2025-08-26 11:31:37.876824341 +0200
+@@ -2394,8 +2394,8 @@
+     void redoWalkBack();
+     //@}
+     
+-    void setMIPStart( const std::vector< std::pair< std::string, double > > &mips ) {
+-       this->mipStart_ = mips;
++    void setMIPStart( const std::vector< std::pair< std::string, double > > &notmips ) {
++       this->mipStart_ = notmips;
+     }
+ 
+     const std::vector< std::pair< std::string, double > > &getMIPStart() {
--- libreoffice-26.2.0.3/external/coinmp/UnpackedTarball_coinmp.mk.vanilla	2026-02-04 13:03:22.564273437 +0100
+++ libreoffice-26.2.0.3/external/coinmp/UnpackedTarball_coinmp.mk	2026-02-04 13:03:33.299051755 +0100
@@ -55,6 +55,7 @@
 	external/coinmp/bind2nd.patch.1 \
 	external/coinmp/clang-with-path.patch \
 	external/coinmp/odr.patch \
+	external/coinmp/mips.patch \
 	external/coinmp/const.patch.1 \
 	external/coinmp/const2.patch.1 \
 	external/coinmp/const3.patch.1 \
